python/sglang/srt/layers/attention/triton_ops/prefill_attention.py

```diff
@@ -104,7 +112,13 @@ def _fwd_kernel(
         qk = tl.zeros([BLOCK_M, BLOCK_N], dtype=tl.float32)
         qk += tl.dot(q, k)
         qk *= sm_scale
-        qk = tl.where(offs_m[:, None] >= (start_n + offs_n[None, :]), qk, float("-inf"))
+        qk += tl.where(
+            (start_n + offs_n[None, :]) < cur_batch_seq_len, 0, float("-inf")
+        )
+        if IS_CAUSAL:
+            qk += tl.where(
+                offs_m[:, None] >= (start_n + offs_n[None, :]), 0, float("-inf")
+            )
```

Comment:
use two branches instead of dong computation twice for IS_CAUSAL

Reply:
Sure.